python/ray/train/tests/test_gpu.py

```diff
@@ -81,9 +86,19 @@ def __getitem__(self, index):
 
 
 # TODO: Refactor as a backend test.
-@pytest.mark.parametrize("num_gpus_per_worker", [0.5, 1])
-def test_torch_get_device(ray_start_4_cpus_2_gpus, num_gpus_per_worker):
+@pytest.mark.parametrize("cuda_visible_devices", ["", "1"])
+def test_torch_get_device(
+    shutdown_only, num_gpus_per_worker, cuda_visible_devices, monkeypatch
+):
+    if cuda_visible_devices:
+        # Test if `get_device` is correct even with user specified env var.
+        monkeypatch.setenv("CUDA_VISIBLE_DEVICES", cuda_visible_devices)
+
+    ray.init(num_cpus=4, num_gpus=2)
+
     def train_fn():
+        if cuda_visible_devices:
+            assert os.environ["CUDA_VISIBLE_DEVICES"] == cuda_visible_devices
```

Comment:
This will always return true right? Or is this just to validate that the environment variable is correctly being set?

Reply:
Yeah this is to validate that it’s being set correctly

the actual test is just to make sure it can run without raising an error. The value of get_device should be the same regardless of what the env car is set to as get_device returns the index.